Std/Data/List/Basic.lean

```diff
@@ -1459,6 +1459,26 @@ zipRight = zipWithRight prod.mk
 -/
 @[inline] def zipRight : List α → List β → List (Option α × β) := zipWithRight Prod.mk
 
+/--
+Version of `List.zipWith` that continues to the end of both lists, passing `none` to one argument
+once the shorter list has run out.
+-/
+def zipWithAll (f : Option α → Option β → γ) : List α → List β → List γ
+  | [], bs => bs.map fun b => f none (some b)
+  | a :: as, [] => (a :: as).map fun a => f (some a) none
+  | a :: as, b :: bs => f a b :: zipWithAll f as bs
```

Comment:
this needs a tail-recursive implementation (or at least a TODO about it)

Reply:
Added a TODO.